geopandas/plotting.py

```diff
@@ -425,19 +425,22 @@ def plot_dataframe(s, column=None, cmap=None, color=None, linewidth=1.0,
                            **color_kwds)
 
     if s[column].dtype is np.dtype('O'):
-        categorical = True
+        try:
+            values = s[column].astype(np.float)
```

Comment:
This was in response to @mlyons-tcc comment in #432 where he suggests to try and cast the data to float before giving up and switching to categorical mode so quickly.  Given that `categorical`  is a parameter it can be confusing to automatically change the value of the parameter.  If the user specifically passed in `categorical` as `False` then they probably don't want a categorical plot.